parallaxScrolling-1.0.js

```diff
@@ -1,153 +1,166 @@
-    } else {
-      var coverWidth = containerWidth,
-          scale = (coverWidth / paramsArr['backgroundWidth']),
-          coverHeight = paramsArr['backgroundHeight'] * scale;
-    }
-
-    paramsArr['backgroundWidth'] = coverWidth;
-    paramsArr['backgroundHeight'] = coverHeight;
-  }
-
-
-  // Calculate the HORIZONTAL background position
-  var xPos = 0;
-  if (scrollRight >= containerLeft && scrollLeft <= containerRight && (paramsArr['maxOffsetX'] != 0 || paramsArr['maxOffsetX'] == 'auto')) {
-    var percentageOfTotalScroll = 100 - (((containerRight - scrollLeft) / containerRight) * 100);
-
-    if (paramsArr['scrollDirectionX'] == 'up') {
-      percentageOfTotalScroll = 100 - percentageOfTotalScroll;
-    }
-    var totalOffset = containerWidth - paramsArr['backgroundWidth'];
-    if (paramsArr['maxOffsetX'] != 'auto' && totalOffset < paramsArr['maxOffsetX']) {
-      totalOffset = paramsArr['maxOffsetX'];
-    }
-    xPos = totalOffset * (percentageOfTotalScroll / 100);
-  }
-
-
-  // Calculate the VERTICAL background position
-  var yPos = 0;
-  if (scrollBottom >= containerTop && scrollTop <= containerBottom && (paramsArr['maxOffsetY'] != 0 || paramsArr['maxOffsetY'] == 'auto')) {
-    var percentageOfTotalScroll = 100 - (((containerBottom - scrollTop) / containerBottom) * 100);
-
-    if (paramsArr['scrollDirectionY']=='up') {
-      percentageOfTotalScroll = 100 - percentageOfTotalScroll;
-    }
-    var totalOffset = containerHeight - paramsArr['backgroundHeight'];
-    if (paramsArr['maxOffsetY'] != 'auto' && totalOffset < paramsArr['maxOffsetY']) {
-      totalOffset = paramsArr['maxOffsetY'];
-    }
-    yPos = totalOffset * (percentageOfTotalScroll / 100);
-  }
-
-  // Apply the background position
-  obj.style.backgroundPosition = (xPos+'px '+yPos+'px');
-}
+(function() {
+  var Parallax = window.Parallax = (function() {
+
+    var self = this;
+    var params = {};
+    var state = {};
+
+    var initialised = false;
+
+    // Init function
+    // ------------------------------------------------
+    // id (str): Required. ID of DOM element.
+    // src (str): Required. Source of background image.
+    // scrollDir (str): Optional. Scroll direction can be either "down" or "up". Default is "down".
+    // maxOffset (int): Optional. Max number of pixels the background can move vertically. By default it will use all available overflowing pixels.
+    this.init = function(id, src, scrollDir, maxOffset) {
+
+      // Make sure it only fires once
+      if (initialised) {
+        return false;
+      }
+
+      // Set optional params
+      var scrollDir = scrollDir || 'down';
+      var maxOffset = (maxOffset * -1) || 'auto';
+
+      // Setup state variables
+      var el = document.getElementById(id);
+      var w = window;
+      var d = document;
+      var e = d.documentElement;
+      var g = d.getElementsByTagName('body')[0]; // For older browsers
+
+      // Add variables to state
+      state = {
+        'el': el,

```

Comment:
```state = {
  el: document.getElementById(id),
```
etc.
No need to assign as variables, then assign to state: just assign to state directly. Also don't make keys strings unless they contain spaces / non-alpha-numeric characters.